presto-main/src/test/java/io/prestosql/sql/query/QueryAssertions.java

```diff
@@ -269,6 +273,11 @@ public QueryAssert matches(@Language("SQL") String query)
         }
 
         private QueryAssert matches(MaterializedResult expected)
+        {
+            return matches(expected, List.of());
+        }
+
+        private QueryAssert matches(MaterializedResult expected, List<Optional<Number>> tolerances)
```

Comment:
on the calling side, it's not clear what these numbers provided to `matches` mean, one needs to lookup parameter name.

Also, i would argue we usually do not need a list, especialy when the result is a single column.
So this could be called like

it would read better as
```
.withNumberTolerance(100)
.matches(MaterializedResult)
```

This would also allow us to provide eg % based tolerance for double values.

(we already have `ordered()` statefulness, this would be similar)

Reply:
Done, modified the code so it will make it explicit that caller needs to specify a percentage tolerance per column